sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IntelliJCacheAccessor.java

```diff
@@ -109,225 +66,50 @@ public String getIntelliJCredentialsFromIdentityMsalCache() {
         return null;
     }
 
-    private String parseRefreshTokenFromJson(String jsonString) {
+    public String parseRefreshTokenFromJson(String jsonString) {
+        /*
+            The json we are parsing looks like this:
+              "RefreshToken": {
+                "rootNode": { // this is an example, this will be some random string related to the account in question.
+                  "home_account_id": "home_account_id",
+                  "environment": "environment",
+                  "client_id": "client_id",
+                  "secret": "refresh_fake_secret",
+                  "credential_type": "credential_type",
+                  "family_id": "family_id"
+                }
+              },
+              so we need to step the parser to find RefreshToken, as noted below.
+         */
         try {
-            JsonNode jsonNode =  DEFAULT_MAPPER.readTree(jsonString);
-            TreeNode refreshTokenNode =  jsonNode.get("RefreshToken");
-            TreeNode baseNode = refreshTokenNode.get(refreshTokenNode.fieldNames().next());
-            TreeNode refreshToken = baseNode.get("secret");
-            String tokenString = refreshToken.toString();
-            if (tokenString.startsWith("\"")) {
-                return tokenString.substring(1, tokenString.length() - 1);
+            try (JsonReader jsonReader = JsonProviders.createReader(jsonString)) {
+                return jsonReader.readObject(reader -> {
+                    while (reader.nextToken() != JsonToken.END_OBJECT) {
+                        String fieldName = reader.getFieldName();
+                        reader.nextToken();
+                        // We only want the "secret" field from the "RefreshToken" node.
+                        if ("RefreshToken".equals(fieldName)) {
+                            reader.nextToken(); // read past the START_OBJECT
+                            reader.nextToken(); // read past the FIELD_NAME for the root sub-object
```

Comment:
Definitely should check this for being the field `rootNode`. I know the expectation is that it will be but we should make sure.

Reply:
"rootNode" here is a placeholder; the actual string is a computed key based on the user's account and some other stuff. The actual json we're reading looks like this:
```
  "RefreshToken": {
    "aguid.anotherguid-login.windows.net-accesstoken-thirdguid-youguesseditanotherguid-openid profile email https://graph.microsoft.com/user.read": {
      "home_account_id": "home_account_id",
      "environment": "environment",
      "client_id": "client_id",
      "secret": "refresh_fake_secret",
      "credential_type": "credential_type",
      "family_id": "family_id"
    }
  }
```
There can in theory be >1 of the key node there. Azure Toolkit uses its own cache file. In the IntelliJ case I think that is probably never true because you don't sign in to Azure Toolkit with more than one account at a time.